File: src/main/java/com/fireblazerrr/survivorbot/utils/message/Messaging.java
package com.fireblazerrr.survivorbot.utils.message;

import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.CommandSender;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

public class Messaging {
    private static final Map<String, ChatColor> colors;

    private Messaging() {

    }

    private static void broadcast(String msg, Object... params) {
        Bukkit.getServer().broadcastMessage(parameterizeMessage(msg, params));
    }

    private static String parameterizeMessage(String msg, Object... params) {
        msg = ChatColor.YELLOW + msg;
        if (params != null) {
            for (int i = 0; i < params.length; ++i) {
                msg = msg.replace("$" + (i + 1), ChatColor.WHITE + params[i].toString() + ChatColor.YELLOW);
            }
        }

        return msg;
    }

    public static ChatColor parseColor(String input) {
        return colors.get(input.toLowerCase().replace("&", ""));
    }

    public static void send(CommandSender sender, String msg, Object... params) {
        sender.sendMessage(parameterizeMessage(msg, params));
    }

    static {
        HashMap<String, ChatColor> tmpMap = new HashMap<>();
        tmpMap.put("black", ChatColor.BLACK);
        tmpMap.put("0", ChatColor.BLACK);
        tmpMap.put("dark blue", ChatColor.DARK_BLUE);
        tmpMap.put("dark_blue", ChatColor.DARK_BLUE);
        tmpMap.put("1", ChatColor.DARK_BLUE);
        tmpMap.put("dark green", ChatColor.DARK_GREEN);
        tmpMap.put("dark_green", ChatColor.DARK_GREEN);
        tmpMap.put("2", ChatColor.DARK_GREEN);
        tmpMap.put("dark aqua", ChatColor.DARK_AQUA);
        tmpMap.put("dark_aqua", ChatColor.DARK_AQUA);
        tmpMap.put("teal", ChatColor.DARK_AQUA);
        tmpMap.put("3", ChatColor.DARK_AQUA);
        tmpMap.put("dark red", ChatColor.DARK_RED);
        tmpMap.put("dark_red", ChatColor.DARK_RED);
        tmpMap.put("4", ChatColor.DARK_RED);
        tmpMap.put("dark purple", ChatColor.DARK_PURPLE);
        tmpMap.put("dark_purple", ChatColor.DARK_PURPLE);
        tmpMap.put("purple", ChatColor.DARK_PURPLE);
        tmpMap.put("5", ChatColor.DARK_PURPLE);
        tmpMap.put("gold", ChatColor.GOLD);
        tmpMap.put("orange", ChatColor.GOLD);
        tmpMap.put("6", ChatColor.GOLD);
        tmpMap.put("gray", ChatColor.GRAY);
        tmpMap.put("grey", ChatColor.GRAY);
        tmpMap.put("7", ChatColor.GRAY);
        tmpMap.put("dark gray", ChatColor.DARK_GRAY);
        tmpMap.put("dark_gray", ChatColor.DARK_GRAY);
        tmpMap.put("dark grey", ChatColor.DARK_GRAY);
        tmpMap.put("dark_grey", ChatColor.DARK_GRAY);
        tmpMap.put("8", ChatColor.DARK_GRAY);
        tmpMap.put("blue", ChatColor.BLUE);
        tmpMap.put("9", ChatColor.BLUE);
        tmpMap.put("bright green", ChatColor.GREEN);
        tmpMap.put("bright_green", ChatColor.GREEN);
        tmpMap.put("green", ChatColor.GREEN);
        tmpMap.put("a", ChatColor.GREEN);
        tmpMap.put("aqua", ChatColor.AQUA);
        tmpMap.put("b", ChatColor.AQUA);
        tmpMap.put("red", ChatColor.RED);
        tmpMap.put("c", ChatColor.RED);
        tmpMap.put("light purple", ChatColor.LIGHT_PURPLE);
        tmpMap.put("light_purple", ChatColor.LIGHT_PURPLE);
        tmpMap.put("pink", ChatColor.LIGHT_PURPLE);
        tmpMap.put("d", ChatColor.LIGHT_PURPLE);
        tmpMap.put("yellow", ChatColor.YELLOW);
        tmpMap.put("e", ChatColor.YELLOW);
        tmpMap.put("white", ChatColor.WHITE);
        tmpMap.put("f", ChatColor.WHITE);
        tmpMap.put("random", ChatColor.MAGIC);
        tmpMap.put("magic", ChatColor.MAGIC);
        tmpMap.put("k", ChatColor.MAGIC);
        tmpMap.put("bold", ChatColor.BOLD);
        tmpMap.put("l", ChatColor.BOLD);
        tmpMap.put("strike", ChatColor.STRIKETHROUGH);
        tmpMap.put("strikethrough", ChatColor.STRIKETHROUGH);
        tmpMap.put("m", ChatColor.STRIKETHROUGH);
        tmpMap.put("underline", ChatColor.UNDERLINE);
        tmpMap.put("n", ChatColor.UNDERLINE);
        tmpMap.put("italic", ChatColor.ITALIC);
        tmpMap.put("o", ChatColor.ITALIC);
        tmpMap.put("reset", ChatColor.RESET);
        tmpMap.put("r", ChatColor.RESET);
        colors = Collections.unmodifiableMap(tmpMap);
    }
}

File: src/main/kotlin/com/fireblazerrr/survivorbot/utils/message/Messaging.kt
package com.fireblazerrr.survivorbot.utils.message

import org.bukkit.ChatColor
import org.bukkit.command.CommandSender

object Messaging {

    private val colors: Map<String, ChatColor> = mapOf(
            "black" to ChatColor.BLACK,
            "0" to ChatColor.BLACK,
            "dark blue" to ChatColor.DARK_BLUE,
            "dark_blue" to ChatColor.DARK_BLUE,
            "1" to ChatColor.DARK_BLUE,
            "dark green" to ChatColor.DARK_GREEN,
            "dark_green" to ChatColor.DARK_GREEN,
            "2" to ChatColor.DARK_GREEN,
            "dark aqua" to ChatColor.DARK_AQUA,
            "dark_aqua" to ChatColor.DARK_AQUA,
            "teal" to ChatColor.DARK_AQUA,
            "3" to ChatColor.DARK_AQUA,
            "dark red" to ChatColor.DARK_RED,
            "dark_red" to ChatColor.DARK_RED,
            "4" to ChatColor.DARK_RED,
            "dark purple" to ChatColor.DARK_PURPLE,
            "dark_purple" to ChatColor.DARK_PURPLE,
            "purple" to ChatColor.DARK_PURPLE,
            "5" to ChatColor.DARK_PURPLE,
            "gold" to ChatColor.GOLD,
            "orange" to ChatColor.GOLD,
            "6" to ChatColor.GOLD,
            "gray" to ChatColor.GRAY,
            "grey" to ChatColor.GRAY,
            "7" to ChatColor.GRAY,
            "dark gray" to ChatColor.DARK_GRAY,
            "dark_gray" to ChatColor.DARK_GRAY,
            "dark grey" to ChatColor.DARK_GRAY,
            "dark_grey" to ChatColor.DARK_GRAY,
            "8" to ChatColor.DARK_GRAY,
            "blue" to ChatColor.BLUE,
            "9" to ChatColor.BLUE,
            "bright green" to ChatColor.GREEN,
            "bright_green" to ChatColor.GREEN,
            "green" to ChatColor.GREEN,
            "a" to ChatColor.GREEN,
            "aqua" to ChatColor.AQUA,
            "b" to ChatColor.AQUA,
            "red" to ChatColor.RED,
            "c" to ChatColor.RED,
            "light purple" to ChatColor.LIGHT_PURPLE,
            "light_purple" to ChatColor.LIGHT_PURPLE,
            "pink" to ChatColor.LIGHT_PURPLE,
            "d" to ChatColor.LIGHT_PURPLE,
            "yellow" to ChatColor.YELLOW,
            "e" to ChatColor.YELLOW,
            "white" to ChatColor.WHITE,
            "f" to ChatColor.WHITE,
            "random" to ChatColor.MAGIC,
            "magic" to ChatColor.MAGIC,
            "k" to ChatColor.MAGIC,
            "bold" to ChatColor.BOLD,
            "l" to ChatColor.BOLD,
            "strike" to ChatColor.STRIKETHROUGH,
            "strikethrough" to ChatColor.STRIKETHROUGH,
            "m" to ChatColor.STRIKETHROUGH,
            "underline" to ChatColor.UNDERLINE,
            "n" to ChatColor.UNDERLINE,
            "italic" to ChatColor.ITALIC,
            "o" to ChatColor.ITALIC,
            "reset" to ChatColor.RESET,
            "r" to ChatColor.RESET
    )

    fun send(sender: CommandSender, msg: String, vararg params: Any) {
        sender.sendMessage(parameterizeMessage(msg, params))
    }

    fun parseColor(input: String): ChatColor {
        return colors.getValue(input.toLowerCase().replace("&", ""))
    }

    private fun parameterizeMessage(msg: String, vararg params: Any): String {
        var message = "${ChatColor.YELLOW} + $msg"
        if (params.isNotEmpty())
            0.until(params.size).forEach { i: Int ->
                message = message.replace("$${i + 1}", "${ChatColor.WHITE} ${params[i]}${ChatColor.YELLOW}")
            }
        return message
    }
}